src/wp-admin/includes/class-wp-list-table.php

```diff
@@ -388,7 +388,15 @@ public function search_box( $text, $input_id ) {
 		$input_id = $input_id . '-search-input';
 
 		if ( ! empty( $_REQUEST['orderby'] ) ) {
-			echo '<input type="hidden" name="orderby" value="' . esc_attr( $_REQUEST['orderby'] ) . '" />';
+			if ( is_array( $_REQUEST['orderby'] ) ) {
+				foreach ( $_REQUEST['orderby'] as $key => $value ) {
+					// Because arrays in query parameters are not standardized, the orderby parameter can be an associative or a non-associative array.
+					$key = is_int( $key ) ? '' : $key;
+					echo '<input type="hidden" name="orderby[' . $key . ']" value="' . esc_attr( $value ) . '" />';
```

Comment:
```suggestion
					echo '<input type="hidden" name="orderby[' . esc_attr( $key ) . ']" value="' . esc_attr( $value ) . '" />';
```

The key needs to be escaped too. In this case it's user input so it's especially important.